lua/customroles/elementalist/sv_elementalist.lua

```diff
@@ -0,0 +1,284 @@
+        return
+    end
+
+    if not att:IsRole(ROLE_ELEMENTALIST) then
+        return
+    end
+
+    -- Att is a valid elementalist damaging a valid player
+    local attId = att:SteamID64()
+    local vicId = ent:SteamID64()
+    local damage = math.Clamp(dmginfo:GetDamage(), 1, 100)
+    local scale = damage * 0.01
+
+    if att:HasEquipmentItem(EQUIP_ELEMENTALIST_FROSTBITE) then
+        local MovementSlow = math.Round(math.Clamp(20 + (scale * 20), 20, 40))
+        local Timer = GetConVar("ttt_elementalist_frostbite_effect_duration"):GetInt()
+
+        if att:HasEquipmentItem(EQUIP_ELEMENTALIST_FROSTBITE_UP) and ChilledPlayers[vicId] and GetChanceConVarOutcome("ttt_elementalist_frostbite+_freeze_chance") then
+            --Upgrade functionality
+            ChilledPlayers[vicId] = false
+            ent:Freeze(true)
+
+            net.Start("BeginIceScreen")
+                net.WriteBool(true)
+            net.Send(ent)
+        else
+            --Base fucntionality
+            ChilledPlayers[vicId] = 1 - (MovementSlow * 0.01)
+            
+            net.Start("BeginIceScreen")
+                net.WriteBool(false)
+            net.Send(ent)
+        end
+
+        local function EndFrostbite()
+            if ent then
+                ChilledPlayers[vicId] = false
+                ent:Freeze(false)
+
+                net.Start("EndIceScreen")
+                net.Start(ent)
+            end
+        end
+
+        if not timer.Exists(vicId .. "_IsSlowed") then
+            timer.Create(vicId .. "_IsSlowed", Timer, 1, EndFrostbite)
+        else
+            timer.Adjust(vicId .. "_IsSlowed", Timer, 1, EndFrostbite)
+        end
+    end
+
+    if att:HasEquipmentItem(EQUIP_ELEMENTALIST_PYROMANCER) then
+        local timerId = vicId .. "_isBurning"
+        local fixTimerId = vicId .. "_IsBurningShotgunFix"
+
+        if att:HasEquipmentItem(EQUIP_ELEMENTALIST_PYROMANCER_UP) and IgnitedPlayers[vicId] and GetChanceConVarOutcome("ttt_elementalist_pyromancer+_explode_chance") then
+            --Upgrade functionality
+            if timer.Exists(fixTimerId) then return end
+
+            timer.Create(fixTimerId, 0.1, 1, function()
+                timer.Remove(fixTimerId)
+            end)
+
+            local explosion = ents.Create("env_explosion")
+
+            if IsValid(explosion) then
+                IgnitedPlayers[vicId] = nil
+                ent:Extinguish()
+
+                explosion:SetPos(ent:GetPos())
+                explosion:SetOwner(att)
+                explosion:Spawn()
+                explosion:SetKeyValue("iMagnitude", ent:Health() * 2)
+                explosion:Fire("Explode", 0, 0)
+                util.BlastDamage(explosion, att, ent:GetPos(), ent:Health() * 4, ent:Health()) 
+            end
+        else
+            --Base functionality
+            IgnitedPlayers[vicId] = true
+            local timerID = vicId .. "_IsBurning"
```

Comment:
Casing is different from usage below
```suggestion
            local timerId = vicId .. "_IsBurning"
```